sqld/src/lib.rs

```diff
@@ -527,7 +534,7 @@ async fn run_storage_monitor(db_path: PathBuf, stats: Stats) -> anyhow::Result<(
             // initialize a connection here, and keep it alive for the entirety of the program. If we
             // fail to open it, we wait for `duration` and try again later.
             let ctx = &mut ();
-            let maybe_conn = match open_db(&db_path, &TRANSPARENT_METHODS, ctx, Some(rusqlite::OpenFlags::SQLITE_OPEN_READ_ONLY)) {
+            let maybe_conn = match open_db(&db_path, &TRANSPARENT_METHODS, ctx, Some(rusqlite::OpenFlags::SQLITE_OPEN_READ_ONLY), DEFAULT_AUTO_CHECKPOINT) {
```

Comment:
Add a comment that this is read only so it won't generate any new frames and the auto_checkpoint setting can be whatever and we just pick DEFAULT_AUTO_CHECKPOINT but it does not matter as no checkpoint is ever performed when no writes appear on the connection.